src/python/pants/core/util_rules/source_files.py

```diff
@@ -66,14 +66,11 @@ async def determine_source_files(request: SourceFilesRequest) -> SourceFiles:
     for hydrated_sources, sources_field in zip(all_hydrated_sources, request.sources_fields):
         if not sources_field.uses_source_roots:
             unrooted_files.update(hydrated_sources.snapshot.files)
-        elif isinstance(sources_field, ResourceSourceField):
+        elif isinstance(sources_field, (ResourceSourceField, ResourcesGeneratingSourcesField)):
             # Resources are special in that they can exist both in and out of source roots, with
             # their source root being optionally stripped.
-            files = list(map(PurePath, hydrated_sources.snapshot.files))
-            osrr = await Get(
-                OptionalSourceRootsResult,
-                SourceRootsRequest(files=files, dirs=()),
-            )
+            files = tuple(PurePath(f) for f in hydrated_sources.snapshot.files)
+            osrr = await Get(OptionalSourceRootsResult, SourceRootsRequest(files=files, dirs=()))
```

Comment:
Can be any iterable, no?

Reply:
Yeah - I specifically changed it because style guide says to not use `map` in general: https://www.pantsbuild.org/v2.9/docs/style-guide#prefer-comprehensions

Reply:
We need to mechanically enforce these or else fast-finger-josh forgets